R/sdk.R

```diff
@@ -0,0 +1,99 @@
+#' Install the Google Cloud SDK
+#'
+#' Installs the Google Cloud SDK which enables CloudML operations.
+#'
+#' @export
+sdk_install <- function() {
+  if (dir.exists(sdk_dir())) {
+    message("SDK already installed.")
+    return(invisible(NULL))
+  }
+
+  if (Sys.info()["sysname"] != "Darwin")
+    stop("Currently, only OS X installation is supported.")
+
+  cloudsdk_url <- "https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/"
+  cloudsdk_tar <- "google-cloud-sdk-178.0.0-darwin-x86_64.tar.gz"
+
+  download_dir <- tempdir()
+  download_file <- file.path(download_dir, cloudsdk_tar)
+
+  if (!dir.exists(download_dir)) dir.create(download_dir)
+
+  download.file(
+    file.path(cloudsdk_url, cloudsdk_tar),
+    download_file
+  )
+
+  extract_dir <- "~/"
+  untar(tarfile = download_file,
+        exdir = extract_dir,
+        tar = "internal")
+}
+
+#' Retrieve Google Cloud SDK Path
+#'
+#' Retrieves the path to the Google Cloud SDK.
+#'
+#' @export
+sdk_dir <- function() {
+  "~/google-cloud-sdk"
```

Comment:
Should we allow users to configure this directory? (I'm okay if we choose not to but some users might not want the SDK in their home directory)

Reply:
Makes sense, configurable using `GCLOUD_INSTALL_PATH`